Sources/SemanticIndex/UpdateIndexStoreTaskDescription.swift

```diff
@@ -104,7 +150,9 @@ public struct UpdateIndexStoreTaskDescription: IndexTaskDescription {
     to currentlyExecutingTasks: [UpdateIndexStoreTaskDescription]
   ) -> [TaskDependencyAction<UpdateIndexStoreTaskDescription>] {
     return currentlyExecutingTasks.compactMap { (other) -> TaskDependencyAction<UpdateIndexStoreTaskDescription>? in
-      guard !other.filesToIndex.intersection(filesToIndex).isEmpty else {
+      guard
+        !other.filesToIndex.hasIntersection(with: filesToIndex, where: { $0.file.mainFile == $1.file.mainFile })
+      else {
```

Comment:
Any concerns about this becoming quadratic as batch sizes grow? Or will the batch sizes always be limited by some constant?

Reply:
I was thinking that building up sets for every `currentlyExecutingTasks` isn’t much cheaper either but on second thought I realized that we can just build up `selfMainFiles` at the start, which seems like it’s beneficial, especially in the worst case.